### PR TITLE
Show an exception info for warnings when -v option given (refs: #7601)

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- refs: https://github.com/sphinx-doc/sphinx/pull/7601#discussion_r419113848

### CODE REVIEW

**Comment by eric-wieser** [comment]
I'm not sure if this approach makes sense. I'd argue you always want to record the exception info, and the flag should just affect the log formatter.

**Comment by eric-wieser** [comment]
Or perhaps argued a different way, I think it should still be opt-in at the log call site, because a lot of exceptions are definitely useless.

So maybe the approach you have here, but have the `Filter` set `exc_info` to `None` instead of having it populate it?

**Comment by tk0miya** [comment]
I thought 1) users would not like to read stacktraces. 2) Only Sphinx and extension developers are needed to read it. So it is enough to show it if requested.

**Comment by eric-wieser** [comment]
I suppose my point is that "showing" stack traces is the job of the `Formatter`, not the job of the logger. If users want to connect up an additional log handler in `conf.py`, they should be given the full information in their `LogRecord`, and be allowed to decide what to do with it themselves.

**Comment by tk0miya** [comment]
Okay, I understand your opinion. I don' have special opinion for the logger's design. So +1 for your idea.

**Comment by tk0miya** [comment]
Updated.

File: sphinx/util/logging.py [hunk 1]
```diff
@@ -131,6 +132,11 @@ def verbose(self, msg: str, *args: Any, **kwargs: Any) -> None:
         self.log(VERBOSE, msg, *args, **kwargs)
 
     def process(self, msg: str, kwargs: Dict) -> Tuple[str, Dict]:  # type: ignore
+        if 'exc_info' not in kwargs:
+            exc_info = sys.exc_info()
+            if exc_info != (None, None, None):
+                kwargs['exc_info'] = True
```
Comment (eric-wieser) [inline sphinx/util/logging.py:138]: I'd still probably argue this is up to the caller to decide, and therefore this change isn't necessary - there are cases when the exception is just not interesting, such as:
```python
try:
    val = some_dict[key]
except KeyError:
    log.info("Unable to find data for {}".format(key))
```
Seeing the exception here is just noise.

File: sphinx/util/logging.py [hunk 2]
```diff
@@ -533,6 +539,19 @@ def format(self, record: logging.LogRecord) -> str:
...
+    def __init__(self, app: "Sphinx") -> None:
+        self.verbosity = app.verbosity
+        super().__init__()
+
+    def formatException(self, ei: Any) -> str:
+        """Show exception info only when verbose mode."""
+        if self.verbosity > 0:
+            return super().formatException(ei)
+        else:
+            return ''
```
Comment (eric-wieser) [inline sphinx/util/logging.py:552]: This looks good, thanks!

File: sphinx/util/logging.py (hunk 1 above)
Comment (tk0miya) [reply sphinx/util/logging.py:138]: I'd like to give exc_info flag to all loggers. But it seems not needed. Okay, withdrawing.

Comment (eric-wieser) [reply sphinx/util/logging.py:138]: I think the rest of this patch is useful, I'd encourage you to withdraw only these four lines.

Comment (tk0miya) [reply sphinx/util/logging.py:138]: I feel too much to add it only for one exc_info call. Do you have a plan to add it more?

**Pull request closed without merging**